### PR TITLE
#615 garden forms

### PR DESCRIPTION
**Pull request by jitka1997**

### CODE REVIEW

**Review by mpinter — COMMENTED** [review]
Makes sense as is to me, would like @MarekBodingerBA to check the schema-generator functions usage.

**Comment by MarekBodingerBA** [comment]
Nie je to https://github.com/bratislava/konto.bratislava.sk/issues/625?

**Review by mpinter — APPROVED** [review]
We can remove files in `pages` before merging. Feel free to keep this open while we iterate small changes on the form and merge once it's ready to proceed.

I've added the form in cli.ts & changed file naming to kebab case to match the others.

File: next/schema-generator/definitions/registracia-predzahradky.ts [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+                'Jarovce',
+                'Karlova Ves',
+                'Lamač',
+                'Nové Mesto',
+                'Petržalka',
+                'Podunajské Biskupice',
+                'Rača',
+                'Rusovce',
+                'Ružinov',
+                'Staré mesto',
```
Comment (MarekBodingerBA) [inline next/schema-generator/definitions/registracia-predzahradky.ts:152]: Staré Mesto

File: next/schema-generator/xmlTemplatePredzahrada.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (MarekBodingerBA) [inline next/schema-generator/xmlTemplatePredzahrada.xml:1]: What is this file for?

File: next/schema-generator/xmlTemplateZahrada.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (MarekBodingerBA) [inline next/schema-generator/xmlTemplateZahrada.xml:1]: What is this file for?

File: next/schema-generator/definitions/registracia-predzahradky.ts [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+  textArea,
+} from '../generator/functions'
+import { createCondition, createStringOptions } from '../generator/helpers'
+
+enum TypRegistracie {
+  NOVA = 'nova',
+  EXISTUJUCA = 'existujuca',
+}
+
+const prilohy = (typRegistracie: TypRegistracie) => [
```
Comment (MarekBodingerBA) [inline next/schema-generator/definitions/registracia-predzahradky.ts:21]: I pushed for product to make the same step for both of the cases, as this is hacky.

File: next/schema-generator/definitions/zalozenie-komunitnej-zahrady.ts [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+                  'Jarovce',
+                  'Karlova Ves',
+                  'Lamač',
+                  'Nové Mesto',
+                  'Petržalka',
+                  'Podunajské Biskupice',
+                  'Rača',
+                  'Rusovce',
+                  'Ružinov',
+                  'Staré mesto',
```
Comment (MarekBodingerBA) [inline next/schema-generator/definitions/zalozenie-komunitnej-zahrady.ts:218]: Staré Mesto

File: next/schema-generator/definitions/zalozenie-komunitnej-zahrady.ts [hunk 6]
```diff
@@ -0,0 +1,373 @@
...
+          },
+        ),
+        conditionalFields(createCondition([[['typPozemku'], { const: 'predschvaleny' }]]), [
+          select(
+            'predschvalenyPozemok',
+            {
+              title: 'Ponuka predschválených mestských pozemkov',
+              required: true,
+              options: [
+                { value: '1', title: 'Azalková' },
```
Comment (MarekBodingerBA) [inline next/schema-generator/definitions/zalozenie-komunitnej-zahrady.ts:165]: Please also use `createStringOptions` for this, this is not stable long term, as "10" or any number can be easily changed for something else when the options are changed.

File: next/schema-generator/definitions/zalozenie-komunitnej-zahrady.ts [hunk 7]
```diff
@@ -0,0 +1,373 @@
...
+                  const: 'iny',
+                },
+              },
+            },
+            required: [],
+          },
+        },
+        required: [],
+      },
+      { title: 'Prílohy', customHash: 'predschvaleny' },
```
Comment (MarekBodingerBA) [inline next/schema-generator/definitions/zalozenie-komunitnej-zahrady.ts:338]: As I mentioned previously, I pushed the product to change this, but I would still like to see "prilohy" here, e.g. "prilohy-predschvaleny-pozemok".

File: next/schema-generator/definitions/zalozenie-komunitnej-zahrady.ts [hunk 8]
```diff
@@ -0,0 +1,373 @@
...
+          },
+          { type: 'dragAndDrop' },
+        ),
+        fileUpload('ine', { title: 'Iné' }, { type: 'dragAndDrop' }),
+      ],
+    ),
+    conditionalStep(
+      'podmienenePrilohyIny',
+      createCondition([[['pozemok', 'typPozemku'], { not: { const: 'predschvaleny' } }]]),
+      { title: 'Prílohy', customHash: 'iny' },
```
Comment (MarekBodingerBA) [inline next/schema-generator/definitions/zalozenie-komunitnej-zahrady.ts:354]: Same here.

File: next/schema-generator/xmlTemplatePredzahrada.xml (hunk 2 above)
Comment (jitka1997) [reply next/schema-generator/xmlTemplatePredzahrada.xml:1]: Committed by accident, I'll remove it.

File: next/schema-generator/xmlTemplateZahrada.xml (hunk 3 above)
Comment (jitka1997) [reply next/schema-generator/xmlTemplateZahrada.xml:1]: Committed by accident, I'll remove it.

**Review by MarekBodingerBA — APPROVED** [review]

**Pull request merged**